From 1c11c626f379fbfd0d1db5dab23dabbfa6dada46 Mon Sep 17 00:00:00 2001
From: Prashanth Dwarakanath <pchengi@nsc.liu.se>
Date: Thu, 21 Jan 2021 14:01:03 +0100
Subject: [PATCH] allow to specify an optional directory into which the
 list-of-modified-files.txt file is written into.

ensure output dir is used if specified.
---
 cmor-fixer.py | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/cmor-fixer.py b/cmor-fixer.py
index fe62b2b..5ff94c1 100755
--- a/cmor-fixer.py
+++ b/cmor-fixer.py
@@ -280,6 +280,8 @@ def main(args=None):
                              "file will be skipped unless the --addatts option is used.")
     parser.add_argument("--olist", "-o", action="store_true", default=False,
                         help="Write list-of-modified-files.txt listing all modified files")
+    parser.add_argument("--output-dir", default=None, type=str,
+                        help="alternate directory to write output file to (optional)")
     parser.add_argument("--addattrs", "-a", action="store_true", default=False,
                         help="Add new attributes from metadata file")
     parser.add_argument("--npp", type=int, default=1, help="Number of sub-processes to launch (default 1)")
@@ -295,6 +297,11 @@ def main(args=None):
         log.error("Options keepid and forceid are mutually exclusive, please choose either one.")
         return
     odir = args.datadir
+    outputdir=args.output_dir
+    if not outputdir == None:
+        if not os.path.isdir(outputdir):
+            log.error("Invalid output directory %s. Skipping run." % outputdir)
+            return
     depth = getattr(args, "depth", None)
     metajson = getattr(args, "meta", None)
     metadata = None
@@ -308,11 +315,15 @@ def main(args=None):
         log.error("Invalid number of subprocesses chosen, please pick a number > 0")
         return
     ofilename = "list-of-modified-files.txt"
+    if not outputdir == None:
+        ofilename="%s/%s"%(outputdir,ofilename)
     if args.olist and os.path.isfile(ofilename):
         i = 1
         while os.path.isfile(ofilename):
             i += 1
             newfilename = "list-of-modified-files-" + str(i) + ".txt"
+            if not outputdir == None:
+                newfilename="%s/list-of-modified-files-"%(outputdir)+str(i)+".txt"
             log.warning("Output file name %s already exists, trying %s..." % (ofilename, newfilename))
             ofilename = newfilename
     if args.npp == 1:
-- 
2.45.2

